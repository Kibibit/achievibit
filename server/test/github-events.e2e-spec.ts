import { INestApplication } from '@nestjs/common';
import { Test, TestingModule } from '@nestjs/testing';
import mongoose from 'mongoose';
import request from 'supertest';

import { PullRequestService } from '@kb-api';
import { AppModule } from '@kb-app';
import { ConfigService } from '@kb-config';
import {
  pullRequestCreatedEvent,
  pullRequestEditedEvent,
  pullRequestLabelAddedEvent,
  pullRequestLabelRemovedEvent,
  pullRequestLabelsInitializedEvent,
  webhookPingEvent
} from '@kb-dev-tools';
import { PullRequest } from '@kb-models';

describe('AppController (e2e)', () => {
  let app: INestApplication;
  let prService: PullRequestService;
  let server;
  let db: typeof mongoose;

  beforeAll(async () => {
    const config = new ConfigService();

    db = await mongoose.connect(config.dbUrl, {
      useFindAndModify: false,
      useNewUrlParser: true,
      useUnifiedTopology: true
    });
  });

  afterAll(async () => {
    await db.connection.db.dropDatabase();
    await db.disconnect();
  });

  beforeEach(async () => {
    await db.connection.db.dropDatabase();
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule]
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
    server = app.getHttpServer();
    prService = moduleFixture.get(PullRequestService);
  });

  afterEach(async () => {
    await app.close();
  });

  test('/ (POST) from github ping event should create db repo', async () => {
    const server = app.getHttpServer();
    const sendWebhookResponse = await request(server)
      .post('/api/webhook-event-manager')
      .send(webhookPingEvent.payload) // x-www-form-urlencoded upload
      .set('Accept', 'application/json')
      .set('x-github-event', webhookPingEvent.event)
      .expect(201);

    expect(sendWebhookResponse.text).toMatchSnapshot();

    const getAllReposResponse = await request(server)
      .get('/api/repo')
      .expect(200);

    expect(getAllReposResponse.body).toMatchSnapshot();
  });

  describe('pr events', () => {
    test('/ (POST) from github pull request created event should create user',
      async () => {
        const server = app.getHttpServer();
        const sendWebhookResponse = await request(server)
          .post('/api/webhook-event-manager')
          .set('Accept', 'application/json')
          .set('x-github-event', pullRequestCreatedEvent.event)
          .send(pullRequestCreatedEvent.payload)
          .expect(201);

        expect(sendWebhookResponse.text).toMatchSnapshot();

        await confirmPrDataCreated();
      });

      test('/ (POST) from github pull request labeled: init', async () => {
        const server = app.getHttpServer();
        await request(server)
            .post('/api/webhook-event-manager')
            .set('Accept', 'application/json')
            .set('x-github-event', pullRequestCreatedEvent.event)
            .send(pullRequestCreatedEvent.payload)
            .expect(201);
        const sendWebhookResponse = await request(server)
          .post('/api/webhook-event-manager')
          .set('Accept', 'application/json')
          .set('x-github-event', pullRequestLabelsInitializedEvent.event)
          .send(pullRequestLabelsInitializedEvent.payload)
          .expect(201);
  
        expect(sendWebhookResponse.text).toMatchSnapshot();
  
        await confirmPrDataCreated();
      });

      test('/ (POST) from github pull request labeled', async () => {
        const server = app.getHttpServer();
        await request(server)
          .post('/api/webhook-event-manager')
          .set('Accept', 'application/json')
          .set('x-github-event', pullRequestCreatedEvent.event)
          .send(pullRequestCreatedEvent.payload)
          .expect(201);
        const sendWebhookResponse = await request(server)
          .post('/api/webhook-event-manager')
          .set('Accept', 'application/json')
          .set('x-github-event', pullRequestLabelAddedEvent.event)
          .send(pullRequestLabelAddedEvent.payload)
          .expect(201);
  
        expect(sendWebhookResponse.text).toMatchSnapshot();
  
        await confirmPrDataCreated();
      });

      test('/ (POST) from github pull request label removed', async () => {
        const server = app.getHttpServer();
        await request(server)
          .post('/api/webhook-event-manager')
          .set('Accept', 'application/json')
          .set('x-github-event', pullRequestCreatedEvent.event)
          .send(pullRequestCreatedEvent.payload)
          .expect(201);
        await request(server)
          .post('/api/webhook-event-manager')
          .set('Accept', 'application/json')
          .set('x-github-event', pullRequestLabelsInitializedEvent.event)
          .send(pullRequestLabelsInitializedEvent.payload)
          .expect(201);
        const sendWebhookResponse = await request(server)
          .post('/api/webhook-event-manager')
          .set('Accept', 'application/json')
          .set('x-github-event', pullRequestLabelRemovedEvent.event)
          .send(pullRequestLabelRemovedEvent.payload)
          .expect(201);
  
        expect(sendWebhookResponse.text).toMatchSnapshot();
  
        await confirmPrDataCreated();
      });

      test('/ (POST) from github pull request edited', async () => {
        const server = app.getHttpServer();
        await request(server)
          .post('/api/webhook-event-manager')
          .set('Accept', 'application/json')
          .set('x-github-event', pullRequestCreatedEvent.event)
          .send(pullRequestCreatedEvent.payload)
          .expect(201);
        const sendWebhookResponse = await request(server)
          .post('/api/webhook-event-manager')
          .set('Accept', 'application/json')
          .set('x-github-event', pullRequestEditedEvent.event)
          .send(pullRequestEditedEvent.payload)
          .expect(201);
    
        expect(sendWebhookResponse.text).toMatchSnapshot();
    
        await confirmPrDataCreated();
      });    

    async function confirmPrDataCreated() {
      const server = app.getHttpServer();
      const getAllUsersResponse = await request(server)
              .get('/api/user')
              .expect(200);
    
      expect(getAllUsersResponse.body).toMatchSnapshot();
    
      const getAllReposResponse = await request(server)
        .get('/api/repo')
        .expect(200);
    
      expect(getAllReposResponse.body).toMatchSnapshot();
    
      const prsDB = await prService.findAllAsync();
      const prs = prsDB.map((prDB) => new PullRequest(prDB.toObject()));
    
      expect(prs).toMatchSnapshot();
    }
  });
});
