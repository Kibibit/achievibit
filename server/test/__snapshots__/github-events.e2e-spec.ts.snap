// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) / (POST) github ping event should create db repo 1`] = `"NewConnection"`;

exports[`AppController (e2e) / (POST) github ping event should create db repo 2`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request assignee added 1`] = `"PullRequestAssigneeAdded"`;

exports[`AppController (e2e) pr events / (POST) github pull request assignee added 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request assignee added 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request assignee added 4`] = `
Array [
  Object {
    "assignees": Array [
      "Thatkookooguy",
    ],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": undefined,
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request assignee removed 1`] = `"PullRequestAssigneeRemoved"`;

exports[`AppController (e2e) pr events / (POST) github pull request assignee removed 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request assignee removed 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request assignee removed 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": undefined,
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request closed 1`] = `"PullRequestClosed"`;

exports[`AppController (e2e) pr events / (POST) github pull request closed 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request closed 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request closed 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": undefined,
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "CLOSED",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request created event should create user 1`] = `"PullRequestOpened"`;

exports[`AppController (e2e) pr events / (POST) github pull request created event should create user 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request created event should create user 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request created event should create user 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": undefined,
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request edited 1`] = `"PullRequestEdited"`;

exports[`AppController (e2e) pr events / (POST) github pull request edited 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request edited 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request edited 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": Object {
      "description": Array [
        null,
      ],
      "title": Array [
        "Create test-event.ts",
      ],
    },
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts - title changed!",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request label removed 1`] = `"PullRequestLabelRemoved"`;

exports[`AppController (e2e) pr events / (POST) github pull request label removed 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request label removed 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request label removed 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": Object {
      "labels": Object {
        "added": 1,
        "removed": 1,
      },
    },
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request labeled 1`] = `"PullRequestLableAdded"`;

exports[`AppController (e2e) pr events / (POST) github pull request labeled 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request labeled 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request labeled 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": Object {
      "labels": Object {
        "added": 1,
      },
    },
    "inlineComments": Array [],
    "labels": Array [
      "pizza",
    ],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request labeled: init 1`] = `"PullRequestInitialLabeled"`;

exports[`AppController (e2e) pr events / (POST) github pull request labeled: init 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request labeled: init 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request labeled: init 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": Object {
      "labels": Object {
        "added": 1,
      },
    },
    "inlineComments": Array [],
    "labels": Array [
      "bug",
    ],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment add 1`] = `"PullRequestReviewCommentAdded"`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment add 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment add 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment add 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": undefined,
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [
      Object {
        "apiUrl": "https://api.github.com/repos/Thatkookooguy/test-new-achievibit-events/pulls/comments/620286741",
        "author": "jaguarBob",
        "commit": "b937b9965d0d8fd3f7ecc2cefbc8fba8931a4622",
        "createdOn": "2021-04-26T13:17:34Z",
        "edited": false,
        "file": "test-event.ts",
        "id": 620286741,
        "message": "this is my review comment!",
        "reviewId": 644659124,
      },
    ],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment del 1`] = `"PullRequestReviewCommentRemoved"`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment del 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment del 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment del 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": Object {
      "reviewComments": Object {
        "deleted": Array [
          620286741,
        ],
      },
    },
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment edit 1`] = `"PullRequestReviewCommentEdited"`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment edit 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment edit 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review comment edit 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": Object {
      "reviewComments": Object {
        "edited": Array [
          620286741,
        ],
      },
    },
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [
      Object {
        "apiUrl": "https://api.github.com/repos/Thatkookooguy/test-new-achievibit-events/pulls/comments/620286741",
        "author": "jaguarBob",
        "body": "this is my review comment!

and now it is edited!",
        "commit": "b937b9965d0d8fd3f7ecc2cefbc8fba8931a4622",
        "createdOn": "2021-04-26T13:17:34Z",
        "edited": true,
        "file": "test-event.ts",
        "id": 620286741,
        "message": "this is my review comment!",
        "reviewId": 644659124,
      },
    ],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review requested 1`] = `"PullRequestReviewRequestAdded"`;

exports[`AppController (e2e) pr events / (POST) github pull request review requested 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review requested 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review requested 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": undefined,
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [
      "norq87",
    ],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review submitted 1`] = `"PullRequestReviewSubmitted"`;

exports[`AppController (e2e) pr events / (POST) github pull request review submitted 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review submitted 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request review submitted 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": undefined,
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [
      Object {
        "authorAssociation": "COLLABORATOR",
        "commit": "b937b9965d0d8fd3f7ecc2cefbc8fba8931a4622",
        "createdOn": "2021-04-26T13:18:03Z",
        "id": 644659124,
        "message": "I requeste these changes please",
        "state": "changes_requested",
        "user": "jaguarBob",
      },
    ],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request reviewer removed 1`] = `"PullRequestReviewRequestRemoved"`;

exports[`AppController (e2e) pr events / (POST) github pull request reviewer removed 2`] = `
Array [
  Object {
    "achievements": Array [],
    "avatar": "https://avatars3.githubusercontent.com/u/10427304?v=4",
    "organization": false,
    "organizations": Array [],
    "repos": Array [],
    "url": "https://github.com/Thatkookooguy",
    "username": "Thatkookooguy",
    "users": Array [],
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request reviewer removed 3`] = `
Array [
  Object {
    "fullname": "Thatkookooguy/test-new-achievibit-events",
    "name": "test-new-achievibit-events",
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events",
  },
]
`;

exports[`AppController (e2e) pr events / (POST) github pull request reviewer removed 4`] = `
Array [
  Object {
    "assignees": Array [],
    "comments": Array [],
    "commits": Array [],
    "createdOn": 2019-12-12T13:48:34.000Z,
    "creator": "Thatkookooguy",
    "description": "",
    "files": Array [],
    "history": Object {
      "deletedReviewers": Array [
        "norq87",
      ],
    },
    "inlineComments": Array [],
    "labels": Array [],
    "number": 1,
    "organization": undefined,
    "prid": "Thatkookooguy/test-new-achievibit-events/pull/1",
    "reactions": Array [],
    "repository": "Thatkookooguy/test-new-achievibit-events",
    "reviewComments": Array [],
    "reviewers": Array [],
    "reviews": Array [],
    "status": "OPEN",
    "title": "Create test-event.ts",
    "updatedDate": undefined,
    "url": "https://github.com/Thatkookooguy/test-new-achievibit-events/pull/1",
  },
]
`;
